frontend/src/context/SearchContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useState, useContext, ReactNode } from 'react';
+
+interface SearchContextType {
+  keyword: string;
+  setKeyword: (keyword: string) => void;
+}
+
+interface SearchProviderProps {
+  children: ReactNode;
+}
+
+const SearchContext = createContext<SearchContextType | undefined>(undefined);
+
+export const SearchProvider: React.FC<SearchProviderProps> = ({ children }) => {
+  const [keyword, setKeyword] = useState<string>('');
```

Comment:
직접 props 타입을 지정하는 방식으로 하는게 좋을 것 같아요!
```suggestion
export const SearchProvider = ({ children } : SearchProviderProps) => {
  const [keyword, setKeyword] = useState<string>('');
```